graveler/committed/merge_iterator.go

```diff
@@ -21,24 +21,31 @@ func NewMergeIterator(diffTheirsToOurs graveler.DiffIterator, base Iterator) (*m
 	return &mergeIterator{diffIt: diffTheirsToOurs, base: base}, nil
 }
 
-func (d *mergeIterator) valueFromBase(key graveler.Key) *graveler.ValueRecord {
+func (d *mergeIterator) valueFromBase(key graveler.Key) (*graveler.ValueRecord, error) {
 	d.base.SeekGE(key)
 	var val *graveler.ValueRecord
 	for d.base.Next() && val == nil {
 		val, _ = d.base.Value()
 	}
+	if err := d.base.Err(); err != nil {
+		return nil, err
+	}
 	if val == nil || !bytes.Equal(val.Key, key) {
-		return nil
+		return nil, nil
 	}
-	return val
+	return val, nil
 }
 
 func (d *mergeIterator) Next() bool {
 	for d.diffIt.Next() {
 		val := d.diffIt.Value()
 		key := val.Key
 		typ := val.Type
-		baseVal := d.valueFromBase(key)
+		baseVal, err := d.valueFromBase(key)
```

Comment:
think we need to set d.value to nil in case of an error or condition Value() if err